presto-spi/src/main/java/com/facebook/presto/spi/function/Signature.java

```diff
@@ -189,7 +151,25 @@ public String toString()
                 longVariableConstraints.stream().map(LongVariableConstraint::toString))
                 .collect(Collectors.toList());
 
-        return name + (allConstraints.isEmpty() ? "" : "<" + Joiner.on(",").join(allConstraints) + ">") + "(" + Joiner.on(",").join(argumentTypes) + "):" + returnType;
```

Comment:
I think you can do `String.join(delimiter, elements)`. Static import `String` if you want.